packages/@aws-cdk/aws-stepfunctions-tasks/lib/apigateway/invoke.ts

```diff
@@ -0,0 +1,277 @@
+        }),
+      };
+    } else if (this.props.headers) {
+      return {
+        Resource: integrationResourceArn('apigateway', 'invoke', this.integrationPattern),
+        Parameters: sfn.FieldUtils.renderObject({
+          ApiEndpoint: this.props.apiEndpoint,
+          Method: this.props.method,
+          Headers: this.props.headers ? this.props.headers.value : sfn.TaskInput.fromDataAt('$.Headers').value,
+          Stage: this.props.stage,
+          Path: this.props.path,
+          AuthType: (this.props.authType ? this.props.authType.value : sfn.TaskInput.fromDataAt('$.AuthType').value) === '$' ? (this.props.authType ? this.props.authType.value : sfn.TaskInput.fromDataAt('$.AuthType').value) : 'NO_AUTH',
+        }),
+      };
+    } else if (this.props.queryParameters) {
+      return {
+        Resource: integrationResourceArn('apigateway', 'invoke', this.integrationPattern),
+        Parameters: sfn.FieldUtils.renderObject({
+          ApiEndpoint: this.props.apiEndpoint,
+          Method: this.props.method,
+          Stage: this.props.stage,
+          Path: this.props.path,
+          QueryParameters: this.props.queryParameters ? this.props.queryParameters.value : sfn.TaskInput.fromDataAt('$.QueryParameters').value,
+          AuthType: (this.props.authType ? this.props.authType.value : sfn.TaskInput.fromDataAt('$.AuthType').value) === '$' ? (this.props.authType ? this.props.authType.value : sfn.TaskInput.fromDataAt('$.AuthType').value) : 'NO_AUTH',
+        }),
+      };
+    } else if (this.props.requestBody) {
+      return {
+        Resource: integrationResourceArn('apigateway', 'invoke', this.integrationPattern),
+        Parameters: sfn.FieldUtils.renderObject({
+          ApiEndpoint: this.props.apiEndpoint,
+          Method: this.props.method,
+          Stage: this.props.stage,
+          Path: this.props.path,
+          RequestBody: this.props.requestBody ? this.props.requestBody.value : sfn.TaskInput.fromDataAt('$.RequestBody').value,
+          AuthType: (this.props.authType ? this.props.authType.value : sfn.TaskInput.fromDataAt('$.AuthType').value) === '$' ? (this.props.authType ? this.props.authType.value : sfn.TaskInput.fromDataAt('$.AuthType').value) : 'NO_AUTH',
+        }),
+      };
+    } else {
+      return {
+        Resource: integrationResourceArn('apigateway', 'invoke', this.integrationPattern),
+        Parameters: sfn.FieldUtils.renderObject({
+          ApiEndpoint: this.props.apiEndpoint,
+          Method: this.props.method,
+          Stage: this.props.stage,
+          Path: this.props.path,
+          AuthType: (this.props.authType ? this.props.authType.value : sfn.TaskInput.fromDataAt('$.AuthType').value) === '$' ? (this.props.authType ? this.props.authType.value : sfn.TaskInput.fromDataAt('$.AuthType').value) : 'NO_AUTH',
+        }),
+      };
+    }
+  }
```

Comment:
there's a lot of duplicated code in this control flow chain from the `if` on down. I think we can simplify this to be a lot more succinct. let's refactor this please!